kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/AbstractBankDocumentService.java

```diff
@@ -0,0 +1,220 @@
+import org.kuali.rice.kew.api.WorkflowDocument;
+import org.kuali.rice.krad.bo.DocumentHeader;
+import org.kuali.rice.krad.rules.rule.event.SaveOnlyDocumentEvent;
+import org.kuali.rice.krad.workflow.service.WorkflowDocumentService;
+import org.kuali.rice.krad.bo.Note;
+import org.kuali.rice.krad.document.Document;
+import org.kuali.rice.krad.service.BusinessObjectService;
+import org.kuali.rice.krad.service.DocumentService;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+
+import java.util.Calendar;
+
+/**
+ * Created by nataliac on 5/3/17.
+ */
+public abstract class AbstractBankDocumentService implements BankDocumentService {
+    private static final Logger LOG = Logger.getLogger(AbstractBankDocumentService.class);
+
+    protected DocumentService documentService;
+    protected WorkflowDocumentService workflowDocumentService;
+    protected TransactionPostingDao transactionPostingDao;
+    protected BankParametersAccessService bankParametersAccessService;
+    protected BusinessObjectService boService;
+    protected AccountingPeriodService accountingPeriodService;
+    protected OptionsService optionsService;
+
+
+    protected abstract KFSConstants.BankTransactionDocumentType getDocumentType();
+
+
+    /**
+     * This method builds the Header of the Document
+     */
+    protected void buildHeader(Document document, BankTransaction bankTransaction) {
+        DocumentHeader documentHeader = document.getDocumentHeader();
+        //set description
+        String documentDescription = StringUtils.isBlank(bankTransaction.getDescriptiveTxt6())?bankTransaction.getBankReference():bankTransaction.getDescription();
+        documentHeader.setDocumentDescription(StringUtils.left(documentDescription, KFSConstants.BankTransactionConstants.MAX_DESCRIPTION_LEN));
+
+        //build explanation string
+        String accountNumber = bankTransaction.getAccountNumber().toString();
+        String baiTypeStr = bankTransaction.getBaiType().toString();
+        String custRefNo = bankTransaction.getCustRefNo();
+        accountNumber = (StringUtils.isBlank(accountNumber)) ? "EMPTY" : accountNumber;
+        accountNumber = StringUtils.left(accountNumber, KFSConstants.BankTransactionConstants.MAX_NUMBER_LEN);
+        custRefNo = (StringUtils.isBlank(custRefNo)) ? "EMPTY" : custRefNo;
+        custRefNo = StringUtils.left(custRefNo, KFSConstants.BankTransactionConstants.MAX_NUMBER_LEN);
+        String explanation = accountNumber + " - " + baiTypeStr + " - " + custRefNo;
+
+        documentHeader.setExplanation(explanation);
+        LOG.debug("buildHeader finished. Header = "+documentHeader.toString());
+    }
+
+    /*//TODO Seems deprecated since all this stuff should be set when the doc is saved and blanketApproved.
+     *  Method implementing a hack to set the CreatedDate, ApprovedDate, FinalizedDate andLastModifiedDate to the given date...
+     */
+    protected void setRoutingDate(Document document, java.sql.Date date) {
```

Comment:
Possible cruft from v3 due to v6 working properly? Doesn't look like this method is part of an interface, so if we are a no-op as is, I'd like to delete the method. Perhaps also make a note in jira, just in case something ever comes up about Date on this.